django/db/backends/mysql/features.py

```diff
@@ -26,6 +26,7 @@ class DatabaseFeatures(BaseDatabaseFeatures):
     can_release_savepoints = True
     atomic_transactions = False
     supports_column_check_constraints = False
+    supports_table_check_constraints = False
```

Comment:
Ah, so MariaDB *does* support them... I guess this'll help push me to get first class MariaDB support into Django after this gets merged.

Also there are some wonderful hacks to make them work as blogged by the MySQL team here: http://mysqlserverteam.com/new-and-old-ways-to-emulate-check-constraints-domain/ . I think the generated column method could potentially be used by Django, it means making very different SQL on MySQL though. Not sure if it's worth it, at least initially.

Reply:
I also think it's not really worth it.